esx_service/cli/vmdkops_admin.py

```diff
@@ -1034,5 +1106,223 @@ def tenant_access_ls(args):
+    if args.datastore and args.local:
+        return err_out("Error: only one of '--datastore' or '--local' can be set")
+    if args.datastore:
+        #TODO: check datastore argument is valid (we already have a code somewhere)
+        pass
+    return None
+
+
+
+def config_init(args):
+    """
+    Init Config DB to allows quotas and access groups (vm-groups)
+    :return: None for success, string for error
+    """
+
+    err = check_ds_local_args(args)
+    if err:
+        return err
+
+    if args.datastore:
+        ds_name = args.datastore
+        db_path = auth_data.AuthorizationDataManager.ds_to_db_path(ds_name)
+        # check that the target datastore is NOT local VMFS, bail out if it is (--force to overide)
+        if is_local_vmfs(ds_name) and not args.force:
+            return err_override("{} is a local datastore.".format(ds_name) +
+                                "Shared datastores are recommended.", "N/A")
+        # Check other datastores, bail out if dockvold/DB exists there.
+        other_ds_config = config_elsewhere(ds_name)
+        if other_ds_config and not args.force:
+            return err_override("Configuration is already inited: {}".format(other_ds_config), "N/A")
+    else:
+        db_path = auth_data.AUTH_DB_PATH
+
+    link_path = auth_data.AUTH_DB_PATH # where was the DB, now is a link
+
+
+    # Check the existing config mode
+    with auth_data.AuthorizationDataManager() as auth:
+        try:
+            auth.connect()
+            if auth.mode == auth_data.DBMode.SingleNode and auth.db_is_empty():
+                print("Removing empty local DB at {}".format(link_path))
+                os.remove(link_path)
+                auth.mode = auth_data.DBMode.NotConfigured
+            info = auth.get_info()
+        except auth_data.DbAccessError as ex:
+            return err_out("Fatal: DB link or DB is corrupted. Check {} ({}".format(link_path, ex))
+
+    if auth.mode  == auth_data.DBMode.NotConfigured:
+        pass
+    elif auth.mode  == auth_data.DBMode.MultiEsx:
+        if not args.force:
+            return err_override("Config DB is already initialized.", info)
+        else:
+            os.remove(link_path)
+    elif auth.mode  == auth_data.DBMode.SingleNode:
+        if not args.force:
+            return err_override("Local DB already exists {}".format(link_path), info)
+        db_move_to_backup(link_path)
+    else:
+        raise Exception("Fatal: Internal error - unknown mode (value %s)" % auth.mode )
+
+
+    if not os.path.exists(db_path):
+        print("Creating new DB at {}".format(db_path))
+        auth = auth_data.AuthorizationDataManager(db_path)
+        err = auth.new_db()
+        if err:
+            return err_out("Init failed: %s" % str(err))
+
+    # Almost done -  just create link and refresh the service
+    if not args.local:
+        print("Creating a symlink to DB at {}".format(link_path))
+        create_db_symlink(db_path, link_path)
```

Comment:
we are already trying to remove a link first. If it failed, it's something else is seriously broken, so create_db_symlink throws an exception and exit.